tavern/internal/redirector/redirector.go

```diff
@@ -0,0 +1,330 @@
+		return
+	}
+
+	if err := stream.CloseSend(); err != nil {
+		handleStreamError(w, "Failed to close gRPC send", err)
+		return
+	}
+
+	setGRPCResponseHeaders(w)
+
+	flusher, ok := getFlusher(w)
+	if !ok {
+		return
+	}
+
+	chunkCount := 0
+	totalBytes := 0
+
+	for {
+		var responseChunk []byte
+		err := stream.RecvMsg(&responseChunk)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			fmt.Printf("[gRPC Stream Error] Failed to receive message: %v\n", err)
+			return
+		}
+
+		chunkCount++
+		totalBytes += len(responseChunk)
+		fmt.Printf("[gRPC Stream] Received chunk %d: %d bytes\n", chunkCount, len(responseChunk))
+
+		// Write gRPC frame header
+		frameHeader := NewFrameHeader(uint32(len(responseChunk)))
+		encodedHeader := frameHeader.Encode()
+		if _, err := w.Write(encodedHeader[:]); err != nil {
+			fmt.Printf("[HTTP Write Error] Failed to write frame header: %v\n", err)
+			return
+		}
+
+		if _, err := w.Write(responseChunk); err != nil {
+			fmt.Printf("[HTTP Write Error] Failed to write chunk: %v\n", err)
+			return
+		}
+
+		flusher.Flush()
+	}
+
+	fmt.Printf("[gRPC -> HTTP] Streamed %d chunks, total %d bytes\n", chunkCount, totalBytes)
+}
+
+func handleReportFileStreaming(w http.ResponseWriter, r *http.Request, conn *grpc.ClientConn) {
+	if !requirePOST(w, r) {
+		return
+	}
+
+	fmt.Printf("[HTTP -> gRPC Client Streaming] Method: /c2.C2/ReportFile\n")
+
+	ctx, cancel := createRequestContext(streamingTimeout)
+	defer cancel()
+
+	stream, err := createStream(ctx, conn, reportFileStream)
+	if err != nil {
+		handleStreamError(w, "Failed to create gRPC stream", err)
+		return
+	}
+
+	buffer := make([]byte, 0, bufferCapacity)
+	chunkCount := 0
+
+	for {
+		readBuf := make([]byte, readChunkSize)
+		n, readErr := r.Body.Read(readBuf)
+		if n > 0 {
+			buffer = append(buffer, readBuf[:n]...)
+		}
+
+		// Process complete gRPC frames from buffer
+		for {
+			header, message, remaining, ok := ExtractFrame(buffer)
+			if !ok {
+				break
+			}
+
+			buffer = remaining
+			chunkCount++
+			fmt.Printf("[Client Stream] Received chunk %d: compression=%d, length=%d bytes\n",
+				chunkCount, header.CompressionFlag, header.MessageLength)
+
+			if err := stream.SendMsg(message); err != nil {
+				handleStreamError(w, "Failed to send gRPC message", err)
+				return
+			}
+		}
+
+		if readErr == io.EOF {
+			break
+		}
+		if readErr != nil {
+			fmt.Printf("[HTTP Read Error] %v\n", readErr)
+			http.Error(w, fmt.Sprintf("Failed to read request body: %v", readErr), http.StatusBadRequest)
+			return
+		}
+	}
+
+	fmt.Printf("[Client Stream] Sent %d chunks total\n", chunkCount)
+
+	if err := stream.CloseSend(); err != nil {
+		handleStreamError(w, "Failed to close gRPC send", err)
+		return
+	}
+
+	var responseBody []byte
+	if err := stream.RecvMsg(&responseBody); err != nil {
+		handleStreamError(w, "Failed to receive gRPC response", err)
+		return
+	}
+
+	fmt.Printf("[gRPC -> HTTP] Response size: %d bytes\n", len(responseBody))
```

Comment:
We don't plan to leave the print statements here right? If we do, we should move this to `slog` like we do elsewhere